components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/restapi/publisher/ApisApiServiceImplUtils.java

```diff
@@ -669,4 +685,266 @@ public static API importAPIDefinition(API apiToAdd, APIProvider apiProvider, Str
 
         return addedAPI;
     }
+
+    /**
+     * @param api API
+     * @param soapOperation SOAP Operation
+     * @return SOAP API Definition
+     * @throws APIManagementException if an error occurred while parsing string to JSON Object
+     */
+    public static String generateSOAPAPIDefinition(API api, String soapOperation) throws APIManagementException {
+
+        APIDefinition oasParser = new OAS2Parser();
+        SwaggerData swaggerData = new SwaggerData(api);
+        String apiDefinition = oasParser.generateAPIDefinition(swaggerData);
+        JSONParser jsonParser = new JSONParser();
+        JSONObject apiJson;
+        JSONObject paths;
+        try {
+            apiJson = (JSONObject) jsonParser.parse(apiDefinition);
+            paths = (JSONObject) jsonParser.parse(soapOperation);
+            apiJson.replace("paths", paths);
+            return apiJson.toJSONString();
+        } catch (ParseException e) {
+            throw new APIManagementException("Error while parsing the api definition.", e);
+        }
+    }
+
+    /**
+     * @param fileInputStream File input stream for the WSDL file
+     * @param url URL
+     * @param wsdlArchiveExtractedPath Path to WSDL extracted directory
+     * @param filename File Name
+     * @return Swagger string
+     * @throws APIManagementException If the WSDL file not supported
+     * @throws IOException If error occurred in converting InputStream to a byte array
+     */
+    public static String getSwaggerString(InputStream fileInputStream, String url, String wsdlArchiveExtractedPath,
+                                          String filename) throws APIManagementException, IOException {
+
+        String swaggerStr = "";
+        if (StringUtils.isNotBlank(url)) {
+            swaggerStr = SOAPOperationBindingUtils.getSoapOperationMappingForUrl(url);
+        } else if (fileInputStream != null) {
+            if (filename.endsWith(".zip")) {
```

Comment:
Shall we use a constant [1] here?

[1] https://github.com/wso2/carbon-apimgt/blob/834c40e6400554a39b7835a45b7ccea681387e2c/apk/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIConstants.java#L1736